Adds a few enhancements to per instance infinite precision

marcusmateus opened the pull request:
This PR should address the following:
* Updates Money constructor to accept an options hash
* Adds `Money.infinite_precision` accessor aliases for backwards compatibility
* When arithmetic is performed between two Money objects preserves the highest precision

**Note**: It is probably easiest to review this PR commit by commit.

Let me know if I anything needs cleanup or if I missed anything.

marcusmateus committed:
Updates Money constructor to accept an options hash

The third parameter of the `Money` constructor now accepts a hash of
options. Currently two options are available:
 - :bank - sets the exchange bank to use with the new instance
 - :infinite_precision - sets the infinite_precision precision for the
     instance, regardless of the global setting

For backwards compatability, if the the third parameters is not a Hash,
it is assumed to be an exchange bank of type `Money::Bank::*`.

marcusmateus committed:
Ensures @infinite_precision instance variable is a boolean

marcusmateus committed:
Aliases Money.infinite_precision for backwards compatability

Aliases Money.infinite_precision accessors to
Money.default_infinite_precision accessors for backwards compatability.
Includes a deprecation notice for those still using the old accessors.

marcusmateus committed:
Arithmetic should preserve the highest precision of operands

marcusmateus commented:
@FunkyloverOne, please take a look at the above when you get a chance.  I think it should get us pretty close to all the way.

FunkyloverOne commented:
Hey @marcusmateus, looks good! I missed out that highest precision preservation, good catch!
I saw you left some commented out `#divmod` related examples. What's up with them?

marcusmateus commented:
Hi @FunkyloverOne, thanks for noticing those commented tests.  I had added them thinking they might be needed. I then realized that they may not really add more value than the tests above and so I commented them out as a reminder to think about it... and then promptly forgot 😞.  That said, the tests do pass if uncommented.

marcusmateus committed:
Removes commented tests from arithmetic spec

They don't really add much value as they are somewhat redundant to the
existing tests.

marcusmateus commented:
Hi @FunkyloverOne, I went ahead and removed those commented tests as I don't really think they added any value.  Let me know if you see anything else that needs addressing.  Thanks!

FunkyloverOne commented:
Just went through it once again, it all looks really good!
I will merge it, so Anthony will be able to review it as well.
